tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitProcessHandler.cs

```diff
@@ -127,54 +127,71 @@ public virtual CommandResult Run(string arguments, string workingDirectory)
-                        {
-                            lock (error)
+                            process.ErrorDataReceived += (s, e) =>
                             {
-                                error.Add(e.Data);
+                                lock (error)
+                                {
+                                    error.Add(e.Data);
+                                }
+                            };
+
+                            process.Start();
+                            process.BeginErrorReadLine();
+                            process.BeginOutputReadLine();
+                            process.WaitForExit();
+
+                            int returnCode = process.ExitCode;
+                            var stdOut = string.Join(Environment.NewLine, output);
+                            var stdError = string.Join(Environment.NewLine, error);
+
+                            DebugLogger.LogDebug($"StdOut: {stdOut}");
+                            DebugLogger.LogDebug($"StdErr: {stdError}");
+                            DebugLogger.LogDebug($"ExitCode: {process.ExitCode}");
+
+                            result.ExitCode = process.ExitCode;
+                            result.StdErr = string.Join(Environment.NewLine, stdError);
+                            result.StdOut = string.Join(Environment.NewLine, stdOut);
+
+                            if (result.ExitCode == 0)
+                            {
+                                break;
                             }
-                        };
-
-                        process.Start();
-                        process.BeginErrorReadLine();
-                        process.BeginOutputReadLine();
-                        process.WaitForExit();
-
-                        int returnCode = process.ExitCode;
-                        var stdOut = string.Join(Environment.NewLine, output);
-                        var stdError = string.Join(Environment.NewLine, error);
-
-                        DebugLogger.LogDebug($"StdOut: {stdOut}");
-                        DebugLogger.LogDebug($"StdErr: {stdError}");
-                        DebugLogger.LogDebug($"ExitCode: {process.ExitCode}");
-
-                        result.ExitCode = process.ExitCode;
-                        result.StdErr = string.Join(Environment.NewLine, stdError);
-                        result.StdOut = string.Join(Environment.NewLine, stdOut);
+                            else if (result.ExitCode != 0)
+                            {
+                                continueToAttempt = IsRetriableGitError(result);
 
-                        if (result.ExitCode != 0)
-                        {
-                            throw new GitProcessException(result);
+                                if(!continueToAttempt)
+                                {
+                                    throw new GitProcessException(result);
+                                }
+                            }
```

Comment:
This could be simplified by just removing the whole else block.

Reply:
I think I got what you were asking for in my latest commit.